lib/mongoid/extensions.rb

```diff
@@ -1,5 +1,17 @@
 # frozen_string_literal: true
 
+class BSON::ObjectId
+  alias :bson_ruby_as_json :as_json
+
+  def as_json(options = nil)
```

Comment:
This method needs a docstring, e.g. copy pasted from the definition in bson-ruby.

Reply:
done

Reply:
`options` is not documented, and why is the signature of this method different from https://github.com/mongodb/bson-ruby/blob/master/lib/bson/object_id.rb#L75 ?

Reply:
I'm not sure I didn't create the method signature... That was created by Durran Jordan 12 years ago. I think I definitely could change it, though, because the * operator is able to take in everything a hash was able to. I don't think that would be a breaking change.

Also options is not documented because options is not used at all in the function. It is also not documented in bson-ruby.  I'm not sure what I would document other than something like "The arguments to the function" without specifying what it's expecting...

Reply:
There are two as_json implementations:

bson-ruby: https://github.com/mongodb/bson-ruby/blob/master/lib/bson/object_id.rb#L75

Mongoid: https://github.com/mongodb/mongoid/blob/7.3-stable/lib/mongoid/extensions.rb#L5

The bson-ruby one is compatible with the Mongoid one, has a docstring and supports more invocation options. Let's use the bson-ruby implementation.

The "This method is intentionally different from #as_extended_json." sentence I think does not make sense in Mongoid's context and should be removed.

Reply:
I removed that part of the comment but I'm not sure what the first point you're trying to make is. The crux of the ticket is to defer to the bson-ruby one, that's what we're adding the feature flag for?

Reply:
I suggest using the bson-ruby signature: `    def as_json(*args)`

Reply:
done